Sprint_09

tayawarmly opened the pull request:
1. Экран пользователя сверстан кодом
2. Добавлен экран SingleImageViewController с использованием ScrollView
3. На экране SingleImageViewController реализована кнопка «Поделиться»

tayawarmly committed:
1. Экран пользователя сверстан кодом
2. Добавлен экран SingleImageViewController с использованием ScrollView
3. На экране SingleImageViewController реализована кнопка «Поделиться»

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let userPicture = UIImage(named: "userPicture")
```
в assets у тебя другое название, не `userPicture`, а `UserPicture`

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        userName.font = .boldSystemFont(ofSize: 23)
+        
+        view.addSubview(userName)
+        view.translatesAutoresizingMaskIntoConstraints = false
```
тут не у view надо проставлять, а у userName. Из-за этого вся верстка едет

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        userNickname.font = .systemFont(ofSize: 13)
+        
+        view.addSubview(userNickname)
+        view.translatesAutoresizingMaskIntoConstraints = false
```
не view, а userNickname

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        userText.numberOfLines = 0
+        
+        view.addSubview(userText)
+        view.translatesAutoresizingMaskIntoConstraints = false
```
не view, а userText

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        exitButton.setImage(UIImage(named: "ExitSymbol"), for: .normal)
+        
+        view.addSubview(exitButton)
+        view.translatesAutoresizingMaskIntoConstraints = false
```
не view, а exitButton

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+            avatarImage.heightAnchor.constraint(equalToConstant: 70)
+        ])
+        
+        let userName = UILabel()
```
лучше все эти вьюхи повыносить в свойства, чтобы можно было их по ходу работы приложения менять. К примеру, в след. спринтах будем получать данные о пользователе с сети, как только данные получили необходимо будет обновить экран. В текущей реализации нормально это сделать не получится, а вот если вынести в свойства, то в любой момент можно будет изменить значение во вьюхах

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        
+        NSLayoutConstraint.activate([
+            userText.leadingAnchor.constraint(equalTo: avatarImage.leadingAnchor),
+            userText.topAnchor.constraint(equalTo: userNickname.bottomAnchor, constant: 8),
```
Тут лучше добавить trailingAnchor, чтобы если текст будет слишком длинным, то он нормально обрезался, а не уходил за пределы экрана
аналогично для userNickname и userName

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        let avatarImage = UIImageView(image: userPicture)
+        self.view.backgroundColor = .ypBlack
+        
+        self.view.addSubview(avatarImage)
```
Тут `self` необязательно использовать, поэтому лучше его убрать.
Подробнее можешь почитать в [код стайле](https://github.com/tutu-ru/swift-style-guide?tab=readme-ov-file#%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8)

helecta commented on ImageFeed/ProfileViewController.swift:
```diff
@@ -0,0 +1,83 @@
+        view.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            exitButton.topAnchor.constraint(equalTo: view.bottomAnchor, constant: 99),
```
не к view.bottomAnchor, а к view.topAnchor, иначе кнопка улетает за пределы экрана

tayawarmly committed:
1. Исправлено отображение страницы профиля
2. Исправлено отображение фотографии профиля
3. Добавлены отступы справа для userName, userNickname и userText
4. Убран лишний отступ на экране ленты

tayawarmly commented:
sprint 09
